addons/stock_picking_batch/views/stock_picking_views.xml

```diff
@@ -16,4 +16,32 @@
             </div>
         </field>
     </record>
+
+    <record id="picking_search_view_inherit" model="ir.ui.view">
+        <field name="name">picking.search.view.inherit</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_internal_search"/>
+        <field name="arch" type="xml">
+            <filter name="my_transfers" position="attributes">
+                <attribute name="domain">
+                    [
+                    '&amp;',
+                    '|', ('batch_id', '=', False), ('batch_user_id', 'in', [uid, False]),
+                    ('user_id', 'in', [uid, False])
+                     ]
```

Comment:
Well it should, but it didn't work, in case of the batch_id is not set, the domain `('batch_user_id', 'in', [uid, False])`  was somehow false.

Reply:
And is it really wanted to select picking where the user AND the batch's user is false or uid ?
Because if a picking is assign to me but is a part of a batch assigned to someone else, I'll not see the picking (which seems to be in contradiction with your other comment and of the user id displayed by the override just below)